core/src/main/java/org/elasticsearch/cluster/routing/RoutingService.java

```diff
@@ -125,6 +132,26 @@ public void clusterChanged(ClusterChangedEvent event) {
                     }
                 }
             }
+
+            long nextDelaySetting = UnassignedInfo.findNextDelayedAllocationSetting(settings, event.state());
```

Comment:
as per discussion can we add a small comment about the rational here? (figure out when the next unassigned allocation need to happen from now. If this is larger or equal then the last time we checked and scheduled, we are guaranteed to have a reroute until then, so no need to schedule again).